src/Shared/InterningBinaryReader.cs

```diff
@@ -7,6 +7,10 @@
 using System.Diagnostics;
 using System.Threading;
 
+#if !NET35
```

Comment:
This is more standard across the codebase

```suggestion
#if !CLR2COMPATIBILITY
```